deploy/quickstart-agw-storage.sh

```diff
@@ -0,0 +1,283 @@
+#! /bin/bash
+
+set -e
+
+if [ -z "$RESOURCE_GROUP" ]; then
+    echo "RESOURCE_GROUP is not set"
+    exit 1
+fi
+if [ -z "$PREFIX" ]; then
+    echo "PREFIX is not set"
+    exit 1
+fi
+if [ -z "$LOCATION" ]; then
+    echo "LOCATION is not set"
+    exit 1
+fi
+if [ -z "$TENANT_ID" ]; then
+    echo "TENANT_ID is not set"
+    exit 1
+fi
+if [ -z "$CERT_PASSWORD" ]; then
+    echo "CERT_PASSWORD is not set"
+    exit 1
+fi
+
+storage_account_name="st$PREFIX"
+vnet_name="vnet-$PREFIX"
+app_gateway_name="gw-$PREFIX"
+ip_stat_name_gateway="ip-gw-$PREFIX"
+iprange_vnet="172.18.0.0/16"
+subnet_name_gateway="subnet-gw-$PREFIX"
+subnet_iprange_gateway="172.18.2.0/24"
+subnet_name_private_link="subnet-pe-$PREFIX"
+subnet_iprange_private_link="172.18.1.0/24"
+private_link_name="pe-$PREFIX"
+public_ip_gw="pip-$PREFIX"
+dns_label="mydomain-$PREFIX"
+
+if [ ! -d "./temp" ]; then
+    mkdir ./temp
+fi
+commonNameFormat="/CN"
+if [[ "$OSTYPE" == "msys" ]]; then
+    commonNameFormat="//CN"
+fi
+
+echo "Starting provisioning of VNET, Storage, App gateway, DNS private records"
+
+# Check if Storage acocunt name is available
+if [ $(az storage account check-name --name $storage_account_name --query nameAvailable) == "false" ]; then
+    echo "Storage account name $storage_account_name is not available, please adapt your PREFIX variable"
+    exit 1
+fi
+
+echo "Creating Resource Group $RESOURCE_GROUP in $LOCATION"
+# Create the Resource group
+az group create --name $RESOURCE_GROUP --location $LOCATION
+
+echo "Creating self-signed test certificate"
```

Comment:
The Key Vault firewall is disabled in this deployment. Can you please enable it after you're done uploading all the certs and add a private endpoint so that App GW can access FW privately and we take Key Vault then off the public internet?